src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ObjectWriter.cs

```diff
@@ -192,31 +199,55 @@ public void BuildSymbolDefinitionMap(ISymbolNode[] definedSymbols)
             }
         }
 
-        public void EmitSymbolDefinition(int currentOffset, NodeFactory factory)
+        private string GetSymbolToEmitForTargetPlatform(string symbol)
+        {
+            string symbolToEmit = symbol;
+
+            if (_targetPlatform.OperatingSystem == TargetOS.OSX)
+            {
+                // On OSX, we need to prefix an extra underscore to account for correct linkage of 
+                // extern "C" functions.
+                symbolToEmit = "_"+symbol;
```

Comment:
The way I look at it - OSX seems to have a rule that all C-style exports should start with an underscore. The question is whether the contract with Objwriter.so/.dll is to emit platform-specific C-style exports (in which case the code should be moved there), or "named exports" in which case we have to deal with it. I don't know how many ways there are to emit C-style exports and whether we want them (eventually) all in ILC.